k9mail/src/main/java/com/fsck/k9/planck/infrastructure/threading/AutoCloseableThread.kt

```diff
@@ -1,12 +1,13 @@
 package com.fsck.k9.planck.infrastructure.threading
 
 open class AutoCloseableThread<T : AutoCloseable>(
-    private val closeable: T,
+    private var closeable: T?,
     target: Runnable?
 ) : Thread(target) {
     override fun run() {
         closeable.use {
             super.run()
         }
+        closeable = null
```

Comment:
is this needed? As far as I know the .use is already taking care of closing the closable, were we concerned about memory leaks? The one thing that might happen is unhandled exceptions, but for these the closable = null will not help.

Reply:
The thing is, the .use closes the closeable, but what I am doing here is freeing the reference to the closeable which is a different thing.